lib/Option/Option.cpp

```diff
@@ -168,7 +179,14 @@ Arg *Option::accept(const ArgList &Args,
     // Matches iff this is an exact match.
     // FIXME: Avoid strlen.
     if (ArgSize != strlen(Args.getArgString(Index)))
+    { // HLSL Change Begin: Except if value separated by spaces here
+      if (JoinedSpaces) {
```

Comment:
If the string provided includes a valid argument followed by extraneous whitespace (e.g. "-external   "), instead of identifying it as an unknown option, this will return a seemingly valid Arg, which might give a confusing error message or take the next argument as the input file.

If JoinedSpaces is set to 0 when the last character in the string is a terminator, then the behavior for this erroneous case should be handled as it was previously.

Reply:
Yeah, that's an odd case.  Here, I think we would end up with an empty string argument value.  I like the idea of resetting the JoinedSpaces if it ended in '\0' to let the user know it was a bad option, as it would have before.

Reply:
So I tried that behavior and it's ugly to try to interpret the error messages that may be produced by this.  It seems easier just to accept and ignore the trailing space case instead.